ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -1156,7 +1155,13 @@ object Serializers {
 
         case TagNew          => New(readClassName(), readMethodIdent(), readTrees())
         case TagLoadModule   => LoadModule(readClassName())
-        case TagStoreModule  => StoreModule(readClassName(), readTree())
+
+        case TagStoreModule =>
+          if (hacks.use13) {
+            readClassName() // className, should always be the enclosing class name
+            readTree() // rhs, should always be This()
+          }
```

Comment:
Can we validate these assumptions? IIRC there is precedent for this kind of "deserialization validation".

I think I'd just much rather have a failure here on unexpected IR than some behavior change in the code.